tensorflow/core/kernels/mkl_matmul_op.cc

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
```

Comment:
This is great. Can you also add version of BatchMatmul based on MKL. Many matmul calls end up in BatchMatmul if the dimensions are not known at graph construction time. See: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/math_ops.py#L1818